src/Corax/Queries/MultiTermMatch.cs

```diff
@@ -100,14 +100,18 @@ public int Fill(Span<long> buffer)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private int FillWithReader(Span<long> buffer)
         {
-            if (_readTerms >= _maxNumberOfTerms)
```

Comment:
This is wrong, talked with @redknightlois and he is testing the proper way to do this.

This code is handling an important optimization for searching by ordered range.